internal/oci/container.go

```diff
@@ -448,13 +460,30 @@ func (c *Container) verifyPid() error {
 }
 
 // getPidStartTime reads the kernel's /proc entry for stime for PID.
-func getPidStartTime(pid int) (int, error) {
-	var st unix.Stat_t
-	if err := unix.Stat(fmt.Sprintf("/proc/%d", pid), &st); err != nil {
-		return 0, errors.Wrapf(ErrNotFound, err.Error())
+// inspiration for this function came from https://github.com/containers/psgo/blob/master/internal/proc/stat.go
+// some credit goes to the psgo authors
+func getPidStartTime(pid int) (string, error) {
+	data, err := ioutil.ReadFile(fmt.Sprintf("/proc/%d/stat", pid))
+	if err != nil {
+		return "", errors.Wrapf(ErrNotFound, err.Error())
+	}
+	// The command (2nd field) is wrapped in ().
+	// Filter it out
+	commEnd := strings.LastIndexByte(string(data), ')')
+	if commEnd == -1 {
+		return "", errors.Wrapf(ErrNotFound, "invalid format in stat file")
+	}
+
+	// commEnd now points to the ending parenthesis.
+	// After it, there is a space, and then the next field starts
+	statFieldsAfterComm := strings.Split(string(data[commEnd+2:]), " ")
+	if len(statFieldsAfterComm) != statTotalFields-statCommField {
+		return "", errors.Wrapf(ErrNotFound, "invalid number of entries found in stat file %d %v", len(statFieldsAfterComm), statFieldsAfterComm)
 	}
 
-	return int(st.Ctim.Sec), nil
+	// and return the 20th
+	// because StartTime is the 22nd field, and we removed the first 2
```

Comment:
You have abstracted these numbers (22 and 2) to constants and then you're hardcoding them here. Yes, it's a comment, but it's still hardcoding :)

Reply:
fixed